charset_normalizer/md.py

```diff
@@ -445,7 +442,7 @@ def ratio(self) -> float:
 
         return self._successive_upper_lower_count_final / self._character_count
 
-
+@lru_cache(maxsize=1024)
```

Comment:
I don't think that is necessary. In theory that should be even slower.

Reply:
This function is the most callable, the effect of adding lru_cache is huge.

without lru_cache ([cachegrind.out.original.zip](https://github.com/Ousret/charset_normalizer/files/8598533/cachegrind.out.original.zip))
![without lru_cache](https://user-images.githubusercontent.com/65482418/166166865-9f0d5e8c-a874-4b8b-a24b-5a0768d40155.jpg)

---
with lru_cache ([cachegrind.out.commits.zip](https://github.com/Ousret/charset_normalizer/files/8598511/cachegrind.out.commits.zip))
![with lru_cache](https://user-images.githubusercontent.com/65482418/166166874-02248017-d746-4069-83a8-3522c6471440.jpg)

Reply:
And for the RAM usage increase?

Reply:
### Memory test (memory-profiler)
ptest.py
```python3
from glob import glob
from os.path import isdir
from charset_normalizer import detect

def performance_compare(size_coeff):
    if not isdir("./char-dataset"):
        print("This script require https://github.com/Ousret/char-dataset to be cloned on package root directory")
        exit(1)
    for tbt_path in sorted(glob("./char-dataset/**/*.*")):
        with open(tbt_path, "rb") as fp:
            content = fp.read() * size_coeff            
        detect(content)

if __name__ == "__main__":
    performance_compare(1)
```
command
```
mprof run ptest.py
mprof plot
```
1. without lru_cache
![1 without lru_cache](https://user-images.githubusercontent.com/65482418/166200690-889b2d56-4e74-4a50-a91d-f53aa91c5d2c.png)

2. lru_cache(is_suspiciously_successive_range)
![2  lru_cache(is_suspiciously_successive_range)](https://user-images.githubusercontent.com/65482418/166200724-cc96c83b-82ff-4860-a814-4d4980340fae.png)

3. lru_cache(is_ascii)
![3  lru_cache(is_ascii)](https://user-images.githubusercontent.com/65482418/166200738-d985d4a0-4e74-49bb-899b-9c49b526413c.png)

4. lru_cache(is_ascii + is_suspiciously_successive_range)
![4  lru_cache(is_ascii + is_suspiciously_successive_range)](https://user-images.githubusercontent.com/65482418/166200756-978cd410-82f1-4b4c-b6b7-bfc2d831610d.png)

Reply:
### size_coeff=100
1. without lru_cache
![100 1  without lru_cache](https://user-images.githubusercontent.com/65482418/166202053-90414d1e-2a33-44a4-85bf-f1301667538e.png)

2. lru_cache(is_ascii + is_suspiciously_successive_range)
![100 4  lru_cache(is_ascii + is_suspiciously_successive_range)](https://user-images.githubusercontent.com/65482418/166202079-97120d67-95ab-4aa4-9606-b04e37a4fdff.png)

Reply:
So the RAM usage does not change much. OK.